modules/rostests/apitests/gdi32/GetDIBits.c

```diff
@@ -452,34 +503,56 @@ void Test_GetDIBits()
     /* Calculate bitmap size and allocate a buffer */
     cjSizeImage = ((pbi->bmiHeader.biWidth * pbi->bmiHeader.biBitCount + 31) / 32) * 4 * pbi->bmiHeader.biHeight;
     ok_int(cjSizeImage, 20);
-    pvBits = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, cjSizeImage);
 
     /* Test using a compatible DC */
-    hdcMem = CreateCompatibleDC(0);
-    ok(hdcMem != 0, "CreateCompatibleDC failed, skipping tests\n");
-    if (hdcMem == NULL) return;
-    ok_int(GetDIBits(hdcMem, hbmp, 0, 4, pvBits, pbi, DIB_RGB_COLORS), 0);
-    ok_int(GetDIBits(hdcMem, ghbmpDIB4, 0, 4, pvBits, pbi, DIB_RGB_COLORS), 4);
+    pvBits = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, cjSizeImage);
+    ok(pvBits != NULL, "HeapAlloc failed\n");
+    if (pvBits == NULL)
+    {
+        skip("No pvBits\n");
+        goto noBits2;
+    }
+
+    hdcMem = CreateCompatibleDC(NULL);
+    ok(hdcMem != NULL, "CreateCompatibleDC failed\n");
+    if (hdcMem == NULL)
+    {
+        skip("No hdcMem\n");
+    }
+    else
+    {
+        ok_int(GetDIBits(hdcMem, hbmp, 0, 4, pvBits, pbi, DIB_RGB_COLORS), 0);
```

Comment:
See comment above. (Concerning these three successive ok() tests.)

Reply:
Wow! You could just write 'apply everywhere', you know...